src/shell.js

```diff
@@ -263,7 +268,14 @@ if (ENVIRONMENT_IS_NODE) {
   };
 
 #if PTHREADS || WASM_WORKERS
-  global.Worker = require('worker_threads').Worker;
+  var worker_threads = require('worker_threads');
+  global.Worker = worker_threads.Worker;
+  ENVIRONMENT_IS_WORKER = !worker_threads.isMainThread;
+#if PTHREADS
+  // Under node we set `workerData` to `em-pthread` to signal that the worker
+  // is hosting a pthread.
+  ENVIRONMENT_IS_PTHREAD = ENVIRONMENT_IS_WORKER && worker_threads['workerData'] == 'em-pthread'
```

Comment:
Does this need to be here, or could it be moved up to where the browser checks `.name`? It does need to be after the `createRequire` code, but that should probably have already been moved before `preJS` anyway.

Reply:
Yes, this block, along with `createRequire` should probably move to before `preJS`..  I think the reason I didn't move that up there already was that it would require creating another `if (ENVIRONMENT_IS_NODE) { .. }` block there which would regress code size.  But I think for correctness it makes sense to move it up there.  I will try it again..

Reply:
That worked, and it only regressed code size by 2 bytes!  Thanks for the suggestion.